src/server/editorServices.ts

```diff
@@ -44,13 +44,18 @@ namespace ts.server {
          * Enum compiler options will be converted to strings.
          */
         readonly compilerOptions: ts.CompilerOptions;
-        // "files", "include", or "exclude" will be undefined if an external config is used.
+        // "extends", "files", "include", or "exclude" will be undefined if an external config is used.
         // Otherwise, we will use "true" if the property is present and "false" if it is missing.
+        readonly extends: boolean | undefined;
         readonly files: boolean | undefined;
         readonly include: boolean | undefined;
         readonly exclude: boolean | undefined;
         readonly compileOnSave: boolean;
         readonly typeAcquisition: ProjectInfoTypeAcquisitionData;
+
+        /** Is 'undefined' if config file name is not tsconfig or jsconfig. */
+        readonly configFileName: "tsconfig.json" | "jsconfig.json" | undefined;

```

Comment:
undefined is not going to survive in the json serialization. consider adding `"other"` value.